docs/data/charts/hooks/hooks.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Charts - Hooks
+productId: x-charts
+---
+
+# Charts - Hooks
+
+<p class="description">The package provides a set of hooks to access chart data and utilities for building custom components.</p>
+
+## Available hooks
+
+The charts package provides several categories of hooks:
+
+### Series hooks
+
+- [**useSeries**](/x/react-charts/hooks/use-series/) - Access raw series data for all chart types
+- Specific series hooks for individual chart types (`useBarSeries`, `useLineSeries`, etc.)
+
+### Legend hooks
+
+- [**useLegend**](/x/react-charts/hooks/use-legend/) - Access formatted legend data for creating custom legend components
+
+### Layout and positioning hooks
+
+- [**useDrawingArea**](/x/react-charts/hooks/use-drawing-area/) - Access the chart's drawing area dimensions and coordinates
+- [**useScale**](/x/react-charts/hooks/use-scale/) - Access D3 scale functions for coordinate transformations (`useXScale`, `useYScale`)
+
+## Quick start
+
+All chart hooks are available from the `@mui/x-charts/hooks` import, with pro and premium packages also providing additional hooks.
+
+```js
+import { useSeries, useLegend, ... } from '@mui/x-charts/hooks';
+import { useSeries, useLegend, ... } from '@mui/x-charts-pro/hooks';
+import { useSeries, useLegend, ... } from '@mui/x-charts-premium/hooks';
+```
+
+## Caveats
+
+All charts hooks require being used within a chart context.
+This means component using those hook should used as:
+
+1. a `slot` of a chart component
+2. a child of a chart component
+3. a child of the `<ChartDataProvider />`
+
+For example if you create a component `<CustomLegend />` that uses the `useLegend()` hook, you could use it as follow:
+
+```jsx
```

Comment:
It would be helpful if the code examples follow the same order as mentioned above:

1. slot
2. child
3. Chart data provider.